UPGRADE_GUIDE.md

```diff
@@ -1,5 +1,34 @@
 # Upgrade Guide
 
+## From `0.3.3` to `0.3.4`
+
+### PlayerLifecycle feature module now provides an EntityId
+
+The callback used for creating a player `EntityTemplate` has changed to provide an `EntityId` up front.
+This will be the `EntityId` the player's Entity will have after it is successfully spawned, and can be usefull for QBI.
```

Comment:
```suggestion
This will be the `EntityId` the player's Entity will have after it is successfully spawned, and can be useful when constructing QBI queries.
```